src/expr/src/vector_op/arithmetic_op.rs

```diff
@@ -327,6 +327,39 @@ where
     r.mul_float(l).ok_or(ExprError::NumericOutOfRange)
 }
 
+#[function("sqrt(float64) -> float64")]
+pub fn sqrt_f64(expr: F64) -> Result<F64> {
+    if expr < F64::from(0.0) {
+        return Err(ExprError::InvalidParam {
+            name: "sqrt input",
+            reason: "input cannot be negative value".to_string(),
+        });
+    }
+    // Edge cases: nan, inf, negative zero should return itself.
+    match expr.is_nan() || expr == f64::INFINITY || expr.is_negative() {
+        true => Ok(expr),
+        false => Ok(expr.sqrt()),
+    }
+}
+
+#[function("sqrt(decimal) -> decimal")]
+pub fn sqrt_decimal(expr: Decimal) -> Result<Decimal> {
+    match expr {
+        Decimal::NaN | Decimal::PositiveInf => Ok(expr),
+        Decimal::Normalized(value) => match value.sqrt() {
```

Comment:
I would like to merge the logic branch so the error is not duplicated.
not sure if its possible to split match branch this
```
Decimal::Normalized(value) if let some(res) = value.sqrt() => {
      ...
}
_ => {
      ...
}
```

Reply:
I am not sure if is possible to match-sqrt-match with a single match either.